src/os/bluestore/BlockDevice.h

```diff
@@ -92,10 +92,10 @@ class BlockDevice {
   uint64_t size;
   uint64_t block_size;
   bool rotational = true;
-  aio_callback_t aio_callback;
-  void *aio_callback_priv;
 
 public:
+  aio_callback_t aio_callback;
+  void *aio_callback_priv;
```

Comment:
If we set task as friend, the patch will become complicated since task is only defined and used in nvmedevice.cc